packages/commonwealth/server/controllers/server_chains_methods/get_related_communities.ts

```diff
@@ -0,0 +1,56 @@
+import { QueryTypes } from 'sequelize';
+import { sequelize } from '../../database';
+import { ServerChainsController } from '../server_chains_controller';
+
+/**
+ * Options for the getRelatedCommunities function.
+ *
+ * @typedef {Object} GetRelatedCommunitiesOptions
+ * @property {string} base - The base variable for filtering (e.g., ChainNodes.name).
+ * @property {string} [searchName] - The optional search term for fuzzy searching chain names (Chains.name).
+ */
+export type GetRelatedCommunitiesOptions = { base: string, searchName?: string };
+
+export type GetRelatedCommunitiesResult = {
+  id: string;
+  community: string;
+  icon_url: string;
+  thread_count: number;
+  address_count: number;
+}[];
+
+export async function __getRelatedCommunities(
+  this: ServerChainsController,
+  { base, searchName }: GetRelatedCommunitiesOptions
+): Promise<GetRelatedCommunitiesResult> {
+  const replacements = searchName ? { base, searchName: `%${searchName}%` } : { base };
+  const filterString = searchName ? 'AND c.name ILIKE :searchName' : ''
+
+  // Although this subquery is not necessary as is currently, We should keep it because in the future if we want to
+  // paginate, then we will need to paginate through the subquery.
+  return await sequelize.query(
+    `
+    SELECT 
+        popular_chains.id as id, 
+        popular_chains.name as community, 
+        popular_chains.icon_url, 
+        popular_chains.thread_count, 
+        COUNT(a) as address_count 
+    FROM 
+        (SELECT c.id, c.icon_url, c.name, COUNT(t) as thread_count 
+        FROM "ChainNodes" as cn 
+        JOIN "Chains" as c on c.chain_node_id = cn.id ${filterString} 
+        LEFT JOIN "Threads" as t on t.chain = c.id 
+        WHERE cn.name = :base 
```

Comment:
I don't like that we use the chain name as the unique identifier field. We should either use chain node id, or the unique identifier provided by eth_chain_id + cosmos_chain_id.